content/articles/storing-data-in-python-using-json-module/index.md

```diff
@@ -0,0 +1,166 @@
+
+Most of your programs will require users to enter various sets of data. Whatever your program's aim is, you'll need data structures like lists and dictionaries to store the data users provide. You will always want to save the data that users enter before they close your program. The best way to do this is to use the JSON module to store your data.
+
+In this tutorial, we'll look at how to store users' data in Python using the JSON module. We shall also learn how to use the JSON module to save and read user-generated data.
+
+### Prerequisites
+Have some basic knowledge of Python programming language.
+
+### Why store data in Python using JSON module?
+
+ 1. Enables developers to dump simple Python data structures to a file and afterward load that data the next time the application runs.
+ 2. Data can be shared between Python programs using Json.
+ 3. When you store data using the JSON format, you can share it even with people working with other programming languages
+ 4. It's simple to learn and comes in a portable format.
+
+### Using json.dump() 
+To use `json.dump()` function import the `json` module first.
+
+The `json.dump()` function take two arguments:
+ 1. Piece of information to save
+ 2. File object it can use to save the information
+
+Let's develop a quick program to save a set of numbers. To save the set of numbers, we will use the `json.dump()` function.
+
+```python
+import json
+
+numbers = [10, 20, 30, 70, 191, 23]  #create a set of numbers
+filename = 'numbers.json'          #use the file extension .json
+with open(filename, 'w') as file_object:  #open the file in write mode
+ json.dump(numbers, file_object)   # json.dump() function to stores the set of numbers in numbers.json file
+```
+
+In this program, we store the set of numbers in the file *numbers.json*. The extension `.json` shows that the data inside the file is in *JSON* form.
+
+We then access the file in `'w'`mode(*write mode*), to make JSON capable of writing data to a file. Finally,the `json.dump()` function keeps the set of numbers in the file *numbers.json*
+
+This program has no terminal output, but when we open the file *numbers.json* we see the following data:
+
+```
+[10, 20, 30, 70, 191, 23]
+```
+
+### Using json.load()
+Let's construct a program that reads the set of numbers back into memory using `json.load()`, considering the previous example:
+
+```python
+import json
+
+filename = 'numbers.json'  #make sure is the same file you wrote to  
+with open(filename,'r') as file_object:  #open in read mode
+ numbers = json.load(file_object)#use json.load to load data and store it in the variable numbers
+
+print(numbers)     #print the set of numbers
+```
+
+The `json.load(`) function takes one argument which is the file object. 
+
+Ensure that you read from the same file that you wrote to. Python only requires to read out of that file hence open it in *read mode*.
```

Comment:
`Ensure that you read from the same file that you wrote to.`

Need not be.

We can load JSON data from any other JSON file too. You should have mentioned to follow the steps in order for getting right results. Only when JSON file is created, it can be loaded.